crypto/s2n_composite_cipher_aes_sha.c

```diff
@@ -125,12 +125,12 @@ static uint8_t s2n_composite_cipher_aes256_sha256_available(void)
 static int s2n_composite_cipher_aes_sha_initial_hmac(struct s2n_session_key *key, uint8_t *sequence_number, uint8_t content_type,
                                                      uint16_t protocol_version, uint16_t payload_and_eiv_len, int *extra)
 {
-    /* BoringSSL does not support these composite ciphers with the existing EVP API, and they took out the
-     * constants used below. This method should never be called with BoringSSL because the isAvaliable checked
-     * will fail. Instead of defining a possibly dangerous default or hard coding this to 0x16 error out with BoringSSL.
+    /* BoringSSL and AWS-LC does not support these composite ciphers with the existing EVP API, and they took out the
+     * constants used below. This method should never be called with BoringSSL or AWS-LC because the isAvaliable checked
+     * will fail. Instead of defining a possibly dangerous default or hard coding this to 0x16 error out with BoringSSL and AWS-LC.
      */
-#ifdef OPENSSL_IS_BORINGSSL
-  S2N_ERROR(S2N_ERR_NO_AVAILABLE_BORINGSSL_API);
+#if defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+  S2N_ERROR(S2N_ERR_NO_AVAILABLE_BSSL_AWSLC_API);
```

Comment:
Is there a better name for this error? It seems unsustainable to just keep adding implementations onto this error name. Maybe name it to indicate the missing functionality instead?

Reply:
Does this better reflect the purpose of the error: `S2N_ERR_NO_SUPPORTED_LIBCRYPTO_API`?